### PR TITLE
api: add EtcdBackup types

### PR DESCRIPTION
**Pull request by fanminshi**

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Commit message and title should be:
```
api: add EtcdBackup types
```

**Comment by fanminshi** [comment]
title changed

File: pkg/apis/etcd/v1beta2/etcd_backup_types.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package v1beta2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type EtcdBackupList struct {
```
Comment (hongchaodeng) [inline pkg/apis/etcd/v1beta2/etcd_backup_types.go]: doc string

Comment (hongchaodeng) [reply pkg/apis/etcd/v1beta2/etcd_backup_types.go]: Same for every other public types

**Comment by fanminshi** [comment]
@hongchaodeng doc string added.

File: pkg/apis/etcd/v1beta2/etcd_backup_types.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	Spec              EtcdBackupSpec   `json:"spec"`
+	Status            EtcdBackupStatus `json:"status,omitempty"`
+}
+
+// EtcdBackupSpec contains a backup specification for an etcd cluster.
+type EtcdBackupSpec struct {
+	// clusterName is the etcd cluster name.
+	ClusterName string `json:"clusterName,omitempty"`
+
+	StorageType string `json:"storageType"`
```
Comment (hongchaodeng) [inline pkg/apis/etcd/v1beta2/etcd_backup_types.go]: doc string

File: pkg/apis/etcd/v1beta2/etcd_backup_types.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	BackupStorageSource `json:",inline"`
```
Comment (hongchaodeng) [inline pkg/apis/etcd/v1beta2/etcd_backup_types.go]: doc string

**Comment by fanminshi** [comment]
@hongchaodeng doc string added.

**Comment by fanminshi** [comment]
@hongchaodeng license header added

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**